File: 4D4Progr4m5/GreedyKnapsack.c
#include<stdio.h>

int main()
{
    float weight[50],profit[50],ratio[50],totalvalue=0,temp,capacity,amount;
    int n,i,j;
    printf("Enter the number of items\n");
    scanf("%d",&n);
    printf("Enter the weights and the profits of the items\n");
    for(i=0;i<n;i++)
    {
        scanf("%f%f",&weight[i],&profit[i]);
    }
    printf("Enter the capacity of knapsack: ");
    scanf("%d",&capacity);
    for(i=0;i<n;i++)
    {
        ratio[i]=profit[i]/weight[i];
    }
    for(i=0;i<n;i++)
    {
        for(j=i+1;j<n;j++)
        {
            if(ratio[i]<ratio[j]){
            temp=ratio[i];
            ratio[i]=ratio[j];
            ratio[j]=temp;

            temp=weight[i];
            weight[i]=weight[j];
            weight[j]=temp;

            temp=profit[i];
            profit[i]=profit[j];
            profit[j]=temp;
            }
        }
    }
    //printf("Solution using Greedy technique is: \n");
    for(i=0;i<n;i++)
    {
        if(weight[i]>capacity)
        {
            break;
        }
        else
        {
            capacity=capacity-weight[i];
            totalvalue=totalvalue+profit[i];
        }
    }
    if(i<n)
    {
        totalvalue=totalvalue+ (ratio[i]*capacity);
    }
    printf("The total value is %f\n",totalvalue);
}

File: 4D4Progr4m5/GreedyKnapsack.py
n = int(input("Enter the number of items : "))
w = input("Enter the weights seperated by space : ")
weights = [int(i) for i in w.split()]
p = input("Enter the profits seperated by space : ")
profits = [int(i) for i in p.split()]
capacity = int(input("Enter the capcaity of the knapsack : "))

ratio = [0.0]*n
for i in range(0,n):
    ratio[i]=profits[i]/weights[i]
    for i in range(0,n):
        for j in range(i+1,n):
            if ratio[i]<ratio[j]:
                temp = ratio[j]
                ratio[j]=ratio[i]
                ratio[i]=temp

                temp=weights[j]
                weights[j]=weights[i]
                weights[i]=temp

                temp=profits[j]
                profits[j]=profits[i]
                profits[i]=temp
totalvalue=0
for i in range(0,n):
    if weights[i]>capacity:
        break
    else:
        totalvalue += profits[i]
        capacity -= weights[i]
if i<n:
    totalvalue = totalvalue + (ratio[i]*capacity)
    print("The maximum value is : ", totalvalue)    